internal/impl/devbox.go

```diff
@@ -722,6 +725,15 @@ func (d *Devbox) computeNixEnv() ([]string, error) {
 	return envPairs, nil
 }
 
+// addConfigEnv takes env variables that are to be set (plugin Env vars)
+// and appends env variables defined in devbox.json to it.
+func (d *Devbox) appendConfigEnv(env []string) []string {
+	for key, value := range d.cfg.Shell.Env {
+		env = append(env, fmt.Sprintf("%s=%s", key, value))
+	}
+	return env
+}
+
```

Comment:
Making *Devbox the receiver here is a bit weird. While it does use a field (`d.cfg.Shell.Env`) the behavior isn't completely obvious to me. I also think if we find ourselves repeating the same logic (get plugin env, get user env) then probably they should live in same function.

Could you instead put this logic into `computeNixEnv` ? 

Side note, I'm not sure if @ipince 's PR is ready, I only see `computeNixEnv` in one place but not in others.

Reply:
`computNixEnv` is used only for devbox run. I spoke with @ipince and he's working consolidating all the env computation and implementations behind feature flags into just one function that is `computeNixEnv` I will then stack my PR on those changes so that the logic can live in `computeNixEnv`